builder/vsphere/iso/step_cleanup.go

```diff
@@ -0,0 +1,36 @@
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type CleanupConfig
+
+package iso
+
+import (
+	"context"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+)
+
+type CleanupConfig struct {
+	// If set to true, the VM will be destroyed after the builder completes
+	Destroy bool `mapstructure:"destroy"`
+}
+
+type StepCleanupVM struct {
+	Config *CleanupConfig
+}
+
+func (c *CleanupConfig) Prepare() []error {
+	return nil
+}
+
+func (s *StepCleanupVM) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+
+	if s.Config.Destroy {
```

Comment:
This exact logic could then be below [line 184](https://github.com/hashicorp/packer-plugin-vsphere/blob/9fc0b63ff62b9460ea15961a04d0fc645e9b520d/builder/vsphere/iso/step_create.go#L184) to ensure the destroy_vm is persisted to state and used in the cleanup.